karrot/users/api.py

```diff
@@ -52,8 +53,12 @@ def get_queryset(self):
 
         groups = self.request.user.groups.all()
         is_applicant_of_group = Q(application__group__in=groups)
+        has_left_group = Q(
+            history__group__in=groups,
+            history__typus=HistoryTypus.GROUP_LEAVE,
```

Comment:
I always struggle to remember exactly what happens when putting multiple deeper (ones with a `__`) conditions in a `Q` query. I think there are some caveats where sometimes it doesn't quite do the intended query and the conditions are not applied to different row or something.

Sorry, I'm a vague on the exact details. I usually end up checking the SQL when I'm not sure (using `print(some_queryset.query)` in a Django shell and pretty printing it, I can understand SQL better than Django ORM :) ) ... also tests are a nice thing too for that.